src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -2235,76 +2252,117 @@ def create_express_route_peering(
-        ResourceType.MGMT_NETWORK,
-        'ExpressRouteCircuitPeering',
-        'ExpressRouteCircuitPeeringConfig',
-        'ExpressRouteCircuitPeeringType',
-        mod='models')
+    (ExpressRouteCircuitPeering, ExpressRouteCircuitPeeringConfig, ExpressRouteCircuitPeeringType,
+     RouteFilter) = get_sdk(
+         ResourceType.MGMT_NETWORK,
+         'ExpressRouteCircuitPeering',
+         'ExpressRouteCircuitPeeringConfig',
+         'ExpressRouteCircuitPeeringType',
+         'RouteFilter',
+         mod='models')
 
-    # TODO: Remove workaround when issue #1574 is fixed in the service
-    # region Issue #1574 workaround
-    circuit = _network_client_factory().express_route_circuits.get(
-        resource_group_name, circuit_name)
-    if peering_type == ExpressRouteCircuitPeeringType.microsoft_peering.value and \
-       circuit.sku.tier == ExpressRouteCircuitSkuTier.standard.value:
-        raise CLIError("MicrosoftPeering cannot be created on a 'Standard' SKU circuit")
-    for peering in circuit.peerings:
-        if peering.vlan_id == vlan_id:
-            raise CLIError(
-                "VLAN ID '{}' already in use by peering '{}'".format(vlan_id, peering.name))
-    # endregion
-
-    peering_config = ExpressRouteCircuitPeeringConfig(
-        advertised_public_prefixes=advertised_public_prefixes,
-        customer_asn=customer_asn,
-        routing_registry_name=routing_registry_name) \
-        if peering_type == ExpressRouteCircuitPeeringType.microsoft_peering.value else None
     peering = ExpressRouteCircuitPeering(
         peering_type=peering_type, peer_asn=peer_asn, vlan_id=vlan_id,
         primary_peer_address_prefix=primary_peer_address_prefix,
         secondary_peer_address_prefix=secondary_peer_address_prefix,
-        shared_key=shared_key,
-        microsoft_peering_config=peering_config)
+        shared_key=shared_key)
+
+    if peering_type == ExpressRouteCircuitPeeringType.microsoft_peering.value:
+        peering.microsoft_peering_config = ExpressRouteCircuitPeeringConfig(
+            advertised_public_prefixes=advertised_public_prefixes,
+            customer_asn=customer_asn,
+            routing_registry_name=routing_registry_name)
     if supported_api_version(ResourceType.MGMT_NETWORK, min_api='2016-12-01') and route_filter:
-        RouteFilter = get_sdk(ResourceType.MGMT_NETWORK, 'RouteFilter', mod='models')
         peering.route_filter = RouteFilter(id=route_filter)
-    return client.create_or_update(
-        resource_group_name, circuit_name, peering_type, peering)
+
+    return client.create_or_update(resource_group_name, circuit_name, peering_type, peering)
+
+
+def _create_or_update_ipv6_peering(config, primary_peer_address_prefix, secondary_peer_address_prefix, route_filter,
+                                   advertised_public_prefixes, customer_asn, routing_registry_name):
+    if config:
+        # update scenario
+        if primary_peer_address_prefix is not None:
```

Comment:
Why not just test:
`if pimary_peer_address_prefix`? Does the empty string have a special meaning?

Reply:
Good point, empty string has no meaning here. Just force of habit.